atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/DegradedClusterInitializationTest.java

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+package com.palantir.cassandra.multinode;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.jayway.awaitility.Awaitility;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfigManager;
+import com.palantir.atlasdb.containers.CassandraContainer;
+import com.palantir.atlasdb.containers.Containers;
+import com.palantir.atlasdb.containers.ThreeNodeCassandraCluster;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueService;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerPort;
+
+public class DegradedClusterInitializationTest {
+
+    private static final String CASSANDRA_NODE_TO_KILL = ThreeNodeCassandraCluster.FIRST_CASSANDRA_CONTAINER_NAME;
+    private static final int CASSANDRA_PORT = ThreeNodeCassandraCluster.CASSANDRA_PORT;
+
+    @ClassRule
+    public static final Containers CONTAINERS = new Containers(DegradedClusterInitializationTest.class)
+            .with(new ThreeNodeCassandraCluster());
+
+    @BeforeClass
+    public static void createKvsAndDegradeCluster() throws IOException, InterruptedException {
+        createCassandraKeyValueService();
+        degradeCassandraCluster();
+    }
+
+    @AfterClass
+    public static void bringClusterBack() throws IOException, InterruptedException {
+        startDeadCassandraNode();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void creatingCassandraKvsShouldFail()  {
+        createCassandraKeyValueService();
+    }
+
+    private static void createCassandraKeyValueService() {
+        CassandraKeyValueService.create(
+                CassandraKeyValueServiceConfigManager.createSimpleManager(ThreeNodeCassandraCluster.KVS_CONFIG),
+                CassandraContainer.LEADER_CONFIG);
+    }
+
+    private static void degradeCassandraCluster() throws IOException, InterruptedException {
+        killFirstCassandraNode();
+
+        // startup checks aren't guaranteed to pass immediately after killing the node, so we wait until
+        // they do. unclear if this is an AtlasDB product problem
+        waitUntilStartupChecksPass();
+    }
+
+    private static void killFirstCassandraNode() throws IOException, InterruptedException {
+        Container container = CONTAINERS.getContainer(CASSANDRA_NODE_TO_KILL);
+        container.kill();
+    }
+
+    private static void startDeadCassandraNode() throws IOException, InterruptedException {
+        Container container = CONTAINERS.getContainer(CASSANDRA_NODE_TO_KILL);
+        container.start();
+        waitUntilCassandraIsListening(container);
+    }
+
+    private static void waitUntilCassandraIsListening(Container container) {
```

Comment:
This doesn't necessarily mean the node is up and running. You should probably check in nodetool